### ISSUE TITLE
Chrome: "This extension contains malware."

### ISSUE THREAD

**Issue by tobiaswaelde** [austencm/youtube-auto-like#61]
I just got a notification from Chrome that this extensions has been disabled because it contains malware.

**Comment by Cerbrus** [comment]
Same issue here, out of nowhere...

**Comment by joaomcarlos** [comment]
Yup, same, if this hasnt been updated in a long time, how does it now contain malware?

**Comment by lollilol** [comment]
youtube removed all auto like youtube plugins, you can see that by googling "youtube auto like", the first 3 extensions all 404
you can probably the extension locally

**Comment by r-jack-ray** [comment]
YouTube lies.  I figure they don't like it messing with metrics.

**Comment by lollilol** [comment]
yep you can still load the extension by just downloading https://github.com/austencm/youtube-auto-like/releases/download/v2.5.2/build.zip, unpacking it into a folder, navigating to chrome://extensions, enabling developer mode and then load an unpacked extension

**Comment by r-jack-ray** [comment]
Thanks.  I was trying to figure out how to get the yarn build working to generate that.

**Comment by MoweME** [comment]
It looks like it's not a false positive.
Some users have been reporting for days about affiliate links suddenly appearing when they go directly to the pages of large stores.

However, I do not guarantee that it is so.

**Comment by Cerbrus** [comment]
Nah the plugin hasn't changed. It looks like Google/Youtube is blocking all auto-like plugins because it apparently messes with their systems...

**Comment by Rgern100** [comment]
I think google just don't want automatic things for like on videos, the extension has not changed for quite a while, the firefox extension still up

**Comment by cstrife1** [comment]
So, consensus is the latest build is safe to use?

**Comment by MacAppSuggestions** [comment]
topic submitted to google devs https://bugs.chromium.org/p/chromium/issues/detail?id=1178256

**Comment by Cerbrus** [comment]
That's not a bug report. That's a unconstructive rant.

**Comment by Delvar** [comment]
I created a quick tamper monkey user script, all i did was copy the code and remove the imports and chrome specific calls.
https://github.com/Delvar/youtube-auto-like-user/blob/main/YouTube%20Auto%20Liker.user.js

maybe this will help some one.

**Comment by austencm** [comment]
Hey guys, sorry about the radio silence. It looks like Google has added new requirements for devs to justify the each permission that their extensions use. They never notified me about this, but I suspect that's the reason for the takedown and malware warning. 

I've filled out the required info and submitted it for review 🤞🏻. Hopefully back up soon. In the meantime, @lollilol is correct. If you really really need it you can download the zip, unzip it, and load the extension unpacked in dev mode.

And if it wasn't already clear: no, there's no malware in here. I would have put some in but I left all my malware in my other pants that day.

**Comment by austencm** [comment]
Update on the takedown situation: I had a frustratingly opaque chat with web store support. They finally admitted that the extension was taken down because it violates YouTube's terms of service. Specifically, the bits about automation and generating fake engagement. Can't really argue with that 😕. Telling people it contained malware was a shitty move, though.

Unfortunately it looks like this is the end of YAL's time as an (official) Chrome extension. I'll keep this repo updated as long as I'm still using it and add some instructions for loading it unpacked. If you use Firefox, the sister extension by @Taknok is still available [here](https://addons.mozilla.org/en-US/firefox/addon/youtube_auto_like/). I doubt that Mozilla will bow to our Google overlords.

**Comment by evgeniy-ugryumov** [comment]
Sorry to hear about the outcome. This is my favorite extension. Like the code too.

**Comment by MacAppSuggestions** [comment]
Like i already wrote before, this is just a bad trend of Google and Google Devs. Is not the first "downgrade" they made. Really bad to have so many extensions simply disabled. In the last 2 years during my hoarding activities i have discovered at least 100+ removed from Chrome Store.

**Comment by Cerbrus** [comment]
> Is not the first "downgrade" they made and they don't fix all related problems generated from such "downgrade" they release, no matter the topic.

Mate, did you read the author's explanation?

Google cracked down on auto-lokers because they violate YouTube's terms. That's not a "downgrade", and Google doesn't "owe" us a fix.   
Whether you agree with those rules or not, it's just Google enforcing their rules.

I'm just happy the plugin still works, as the content creators do deserve the likes, even if we forget to click'm, some times.

**Comment by Taknok** [comment]
I hope so @austencm , Alphabet (Google mother company) is supporting Mozzilla quite a lot :/
On Firefox the "store" AMO is not force for app. Here is the [Mozilla doc](https://extensionworkshop.com/documentation/manage/updating-your-extension/) I did not looked, but there may be a similar way for Chrome ?

**Comment by MacAppSuggestions** [comment]
This is just an excuse for this extensions. What about all other 30 they simply removed in the last months and have nothing to do with youtube auto like? They will probably find another excuse when you contact them.

If this would be true, they should ban all extensions https://chrome.google.com/webstore/search/auto%20like%20youtube?_category=extensions no matter if they work or not, since the topic is the same; and not just this extension. Plus if this is true, the should ban all extensions (no matter the browser....), because has nothing to do with Google policies, but with YouTube policies. 

Again, the can even implement something like Reddit, where they disable such feature for the public, but you can still continue to like and hoard videos. 

Another option is to enable auto manual installed extension. Last option is to avoid to deactivate it, if we already installed it. Now i'm talking about extension in general, and not only about ytautolike

**Comment by austencm** [comment]
@Taknok I don't of anything like that in Chrome but I could probably have it ping Github to check for updates. If there's enough people requesting it I'd try it but I doubt there are more than like 20 people that have this installed right now lol.

@MacAppSuggestions I agree with you that the web store has been kind of a shitshow. With manifest v3 (update to Chrome's extension API) coming out it appears that Google is finally giving extensions more attention and I suspect the takedowns are part of that. Cleaning house.

It's true there are other extensions still up that for sure violate YT's terms in different ways. Frustrating that they enforce their rules selectively, but at the same time I don't want those other extensions taken down. I have one that hides videos by channel and I'll be devastated if I have to see WatchMojo in my feed again.

**Comment by markygnlg** [comment]
4 things you might want to test/try:
'
- there is now a v2.7.1 which was released about 24hrs ago as of this writing:
https://github.com/austencm/youtube-auto-like/releases
'
- secondly, I too found it 'not working' (with the 50% setting) but if I pressed 'F5'/refresh or opened the video in another tab (same as refreshing the page) the like button would be lit up: ie. it works but the graphics of the thumb only displays if you press refresh (I'm using v2.7.1 on Opera)
'
- try testing with 'Instantly'
'
- updates default to "Only videos from subbed channels" - it's working for me on all videos after I changed the setting to "ALL videos"
'
P.S.
 Sadly, Chrome removed this extension for violating YouTube's terms of service (specifically the part about causing inaccurate measurements of user engagement). [from https://github.com/austencm/youtube-auto-like ... or click on the like "CODE" at the top of the page]
